File: 263.cpp
// https://leetcode.com/problems/ugly-number/
class Solution {
 public:
  bool isUgly(int num) {
    auto div = [](int& val, int to_div) {
      while (val && val % to_div == 0) {
        val /= to_div;
      }
    };
    div(num, 2);
    div(num, 3);
    div(num, 5);
    return num == 1;
  }
};

File: golang/263.go
package main

func isUgly(num int) bool {
	div := func(val *int, to_div int) {
		for {
			if *val == 0 || *val%to_div != 0 {
				break
			}
			*val /= to_div
		}
	}
	div(&num, 2)
	div(&num, 3)
	div(&num, 5)

	return num == 1
}
